cpp/ql/lib/semmle/code/cpp/models/implementations/ODBC.qll

```diff
@@ -0,0 +1,30 @@
+/**
+ * Provides implementation classes modeling the ODBC C/C++ API.
+ * See `semmle.code.cpp.models.Models` for usage information.
+ */
+
+private import semmle.code.cpp.models.interfaces.Sql
+private import semmle.code.cpp.models.interfaces.FunctionInputsAndOutputs
+
+/**
+ * The `SQLExecDirect`, and `SQLPrepare` from the ODBC C/C++ API:
+ * https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlexecdirect-function?view=sql-server-ver16
+ * https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlprepare-function?view=sql-server-ver16
+ *
+ * Note, `SQLExecute` is not included because it operates on a SQLHSTMT type, not a string.
+ * The SQLHSTMT parameter for `SQLExecute` is set through a `SQLPrepare`, which is modeled.
+ * The other source of input to a `SQLExecute` is via a `SQLBindParameter`, which sanitizes user input,
+ * and would be considered a barrier to SQL injection.
+ */
+private class ODBCExecutionFunction extends SqlExecutionFunction {
+  ODBCExecutionFunction() {
+    exists(string s | s in ["SQLExecDirect", "SQLPrepare"] and this.hasName(s))
+  }
```

Comment:
Two changes:
- No need to introduce a `string`. Let's just inline the set literal (which is what we call the `[a, b, c]` syntax).
- Replace `hasName` with `hasGlobalName`. I don't think we want to match member functions named `SQLExecDirect` or `SQLPrepare`, right? The two functions we're interested are non-member functions in the global namespace, right?

```suggestion
  ODBCExecutionFunction() { this.hasGlobalName(["SQLExecDirect", "SQLPrepare"]) }
```

Reply:
Done